polygeist/tools/cgeist/driver.cc

```diff
@@ -316,7 +316,9 @@ static LogicalResult canonicalize(mlir::MLIRContext &Ctx,
   OptPM.addPass(polygeist::createRemoveTrivialUsePass());
   OptPM.addPass(polygeist::createMem2RegPass());
   OptPM.addPass(mlir::createCanonicalizerPass(CanonicalizerConfig, {}, {}));
-  OptPM.addPass(polygeist::createLoopRestructurePass());
+  // FIXME: This is causing infinite recursion on host code.
+  if (SYCLUseHostModule == "")
+    OptPM.addPass(polygeist::createLoopRestructurePass());
```

Comment:
Can we at least run this pass on the gpu module ? Or do you think is easier to fix the issue causing infinite recursion (in a future PR) and live with this for now ?

Reply:
Ok